### PR TITLE
Add possiblity to remove maximum size limit of file-backed pmem kind

### PR DESCRIPTION
**Pull request by michalbiesek**
This change is [Reviewable](https://reviewable.io/reviews/memkind/memkind/60)

### CODE REVIEW

File: man/memkind.3 [hunk 1]
```diff
@@ -258,15 +258,20 @@ on a call to
 .BR memkind_pmem (3)).
 This allows to create a pmem memkind of a pretty large size without the
 need to reserve in advance the corresponding file system space for the entire
-heap.  The minimum
+heap. If the value of
 .I max_size
-value allowed by the library is defined in
+is equal 0, pmem memkind is not limited by the file system specified by the
```
Comment (pbalcer) [inline man/memkind.3]: pmem memkind is only limited by the capacity of the file system mounted under *dir*.

Comment (michalbiesek) [reply man/memkind.3]: Done.

**Comment by kasiawasiuta** [comment]
retest this please

**Comment by michalbiesek** [comment]
___
*[man/memkind.3, line 263 at r2](https://reviewable.io/reviews/memkind/memkind/60#-LIpnT0eDWTBlsI9ecmQ:-LItGgyx3HPgtjOXxO1M:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/41dbf3fd09285941e1b2fb14a3960bf53143ddfb/man/memkind.3#L263)):*
<details><summary><i>Previously, KFilipek (Krzysztof Filipek) wrote…</i></summary><blockquote>

"is equal to 0" or "equals 0"
</blockquote></details>

Done.

**Comment by michalbiesek** [comment]
___
*[src/memkind_pmem.c, line 160 at r3](https://reviewable.io/reviews/memkind/memkind/60#-LIu3kTJ4CtylD3xG5de:-LIuFQnFDdHvDvAxMfxH:b949ojq) ([raw file](https://github.com/memkind/memkind/blob/27b26092679817e3258f539f9e1ebc861ec304d4/src/memkind_pmem.c#L160)):*
<details><summary><i>Previously, KFilipek (Krzysztof Filipek) wrote…</i></summary><blockquote>

Memkind_pmem_create have different formatting
</blockquote></details>

Yes I know is different, I am aware of differences. I keep it consistent with other extent_hooks_s in this file. 
I plan to perform separate PR, which will format all files according to CODING STYLE section described in CONTRIBUTING.

**Comment by michalbiesek** [comment]
___
*[src/memkind_pmem.c, line 249 at r3](https://reviewable.io/reviews/memkind/memkind/60#-LIuDQBB2azWPEFo5hZT:-LIufEr5-F_QhdMqpbUM:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/27b26092679817e3258f539f9e1ebc861ec304d4/src/memkind_pmem.c#L249)):*
<details><summary><i>Previously, KFilipek (Krzysztof Filipek) wrote…</i></summary><blockquote>

Maybe != MAP_FAILED, code will look like:
if(assignment != MAP_FAILED) {
priv->offset += size; 
}
rest without change and code duplication (result will have error code)
</blockquote></details>

Done.

**Pull request merged**